paddle/fluid/framework/details/var_handle.h

```diff
@@ -66,6 +66,11 @@ struct VarHandle : public VarHandleBase {
     return o.generated_op_ == generated_op_ && o.name_ == name_ &&
            o.scope_idx_ == scope_idx_;
   }
+
+  bool operator!=(const VarHandle& o) const {
+    return o.generated_op_ != generated_op_ || o.name_ != name_ ||
+           o.scope_idx_ != scope_idx_;
```

Comment:
what about version?

Reply:
Thanks, this operator is used in [here](https://github.com/PaddlePaddle/Paddle/pull/10096/files/7ee07df7ad06666d5e31e520953185df04384644#diff-01be9a0176c608e1de2269e08d2ea181R89), 
```
if (*out != *in_var_handle) 
```
The `version_` of them can be different, but the `place_`.....
Let me think about it again.

Reply:
It is no need to implement `!=` manually. You can just invoke `return this->operator==(o);`